src/language/worker.ts

```diff
@@ -74,168 +67,46 @@ const executeAll = async (
         }
       },
     });
+
+    // Evaluate a playground content.
     const { source, type, printable } = await runtime.evaluate(codeToEvaluate);
     const namespace =
       source && source.namespace
         ? `${source.namespace.db}.${source.namespace.collection}`
         : null;
-    const result: PlaygroundResult = {
+
+    // Prepare a playground result.
+    const result = {
       namespace,
       type: type ? type : typeof printable,
       content: getContent({ type, printable }),
       language: getLanguage({ type, printable }),
     };
 
-    return [null, { outputLines, result }];
-  } catch (error) {
-    return [error];
-  }
-};
-
-const findAndParse = async (
-  serviceProvider: CliServiceProvider,
-  databaseName: string,
-  collectionName: string
-) => {
-  const documents = await serviceProvider
-    .find(databaseName, collectionName, {}, { limit: 1 })
-    .toArray();
-
-  if (documents.length === 0) {
-    return [];
-  }
-
-  try {
-    const schema = await parseSchema(documents);
-
-    if (!schema || !schema.fields) {
-      return [];
-    }
-
-    const fields = schema.fields.map((item) => ({
-      label: item.name,
-      kind: CompletionItemKind.Field,
-    }));
-
-    return fields;
-  } catch (parseError) {
-    return [];
-  }
-};
-
-const getFieldsFromSchema = async (
-  connectionString: string,
-  connectionOptions: any,
-  databaseName: string,
-  collectionName: string
-): Promise<any> => {
-  try {
-    const serviceProvider: CliServiceProvider =
-      await CliServiceProvider.connect(connectionString, connectionOptions);
-
-    const result = await findAndParse(
-      serviceProvider,
-      databaseName,
-      collectionName
-    );
-
-    return [null, result];
-  } catch (error) {
-    return [error];
-  }
-};
-
-const prepareCompletionItems = (result: Document) => {
-  if (!result) {
-    return [];
-  }
-
-  return result.databases.map((item) => ({
-    label: item.name,
-    kind: CompletionItemKind.Value,
-  }));
-};
-
-const getListDatabases = async (
-  connectionString: string,
-  connectionOptions: any
-) => {
-  try {
-    const serviceProvider: CliServiceProvider =
-      await CliServiceProvider.connect(connectionString, connectionOptions);
-
-    // TODO: There is a mistake in the service provider interface
-    // Use `admin` as arguments to get list of dbs
-    // and remove it later when `mongosh` will merge a fix.
-    const result = await serviceProvider.listDatabases('admin');
-    const databases = prepareCompletionItems(result);
-
-    return [null, databases];
+    return { data: { outputLines, result } };
   } catch (error) {
-    return [error];
+    return { error };
```

Comment:
I think depcheck was failing because this was previously `throw new Error((<any>error).message);`, and the parser it uses didn’t understand the `<any>error` cast – you should be able to remove the addition to `.depcheckrc` now that that’s no longer the case

Reply:
You are right!! 🔥